### PR TITLE
Extract project with reusable elements

### PR DESCRIPTION
**Pull request by meaboutsoftware**
This PR contains work related to:

- Creation of ReusableElements project that contains building blocks for each module
- Move of ApiPaths to this project
- Move of SystemClock to this project

### CODE REVIEW

File: Chapter-2-modules-separation/Src/Fitnet.IntegrationTests/Common/TestEngine/SystemClock/FakeSystemClock.cs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 namespace EvolutionaryArchitecture.Fitnet.IntegrationTests.Common.TestEngine.SystemClock;
 
-using EvolutionaryArchitecture.Fitnet.Shared.SystemClock;
+using ReusableElements.SystemClock;
```
Comment (kamilbaczek) [inline Chapter-2-modules-separation/Src/Fitnet.IntegrationTests/Common/TestEngine/SystemClock/FakeSystemClock.cs:3]: This is my personal opinion. I don't like "ReusableElements" such a name will encourage you to put there everything that does not fit into functional projects. I would suggest introducing categorizations e.g. Shared. Infrastructure and Clock will be there. 

We could have another Shared.DomainDrivenDesign.BuildingBlock in the future.
Api path could be in Shared.Api
Such categorization will navigate developer where should put reusable.
What you think?

Comment (meaboutsoftware) [reply Chapter-2-modules-separation/Src/Fitnet.IntegrationTests/Common/TestEngine/SystemClock/FakeSystemClock.cs:3]: I do not like Shared 😜 I will think about the change.

Yes, I thought about the structure like:

Xyz.Api
Xyz.Infrastructure
Xyz.Application
Xyz.Domain
Xyz.Tests (things reusable for tests if we have something that can be reused)

in the next step but we can already start with:

Xyz.Api (here goes ApiPaths)
Xyz.Infrastructure (SystemClock)

Wdyt?

Maybe Blocks.Api, Blocks.Infrastructure?

Comment (kamilbaczek) [reply Chapter-2-modules-separation/Src/Fitnet.IntegrationTests/Common/TestEngine/SystemClock/FakeSystemClock.cs:3]: Great. Maybe let's stay with you suggestion and we will spend some time on our status meeting to look for better name.

Comment (meaboutsoftware) [reply Chapter-2-modules-separation/Src/Fitnet.IntegrationTests/Common/TestEngine/SystemClock/FakeSystemClock.cs:3]: I named it common

**Review by kamilbaczek — APPROVED** [review]
Great job 💪👍

**Pull request merged**